--- a/libsrc/Dk/Dkmarshal.c	2020-12-23 15:53:32.000000000 +0900
+++ b/libsrc/Dk/Dkmarshal.c	2020-12-23 15:57:40.000000000 +0900
@@ -31,25 +31,14 @@
 # include <netinet/in.h>			 /* for ntohl, htonl */
 #endif
 
-#if defined (i386) || \
-    defined (_WIN64) || \
-    defined (_M_IX86) || \
-    defined (_M_ALPHA) || \
-    defined (mc68000) || \
-    defined (sparc) || \
-    defined (__x86_64) || \
-    defined (__alpha) || \
-    defined (__powerpc) || \
-    defined (mips) || \
-    defined (__OS2__) || \
-    defined (_IBMR2)
-# define _IEEE_FLOATS
-#elif defined (OPL_SOURCE)
+#ifndef _IEEE_FLOATS
+#if defined (OPL_SOURCE)
 # include <librpc.h>
 #else
 # include <rpc/types.h>
 # include <rpc/xdr.h>
 #endif
+#endif
 
 macro_char_func readtable[256];
 ses_write_func writetable[256];
--- a/libsrc/Dk.h	2018-08-16 08:06:21.000000000 +0900
+++ b/libsrc/Dk.h	2020-10-22 18:25:57.000000000 +0900
@@ -8,7 +8,7 @@
  *  This file is part of the OpenLink Software Virtuoso Open-Source (VOS)
  *  project.
  *  
- *  Copyright (C) 1998-2018 OpenLink Software
+ *  Copyright (C) 1998-2020 OpenLink Software
  *  
  *  This project is free software; you can redistribute it and/or modify it
  *  under the terms of the GNU General Public License as published by the
@@ -29,6 +29,20 @@
 #ifndef _DK_H
 #define _DK_H	/* libutil needs this name !!! */
 
+#ifndef VIRT_C_LINKAGE
+#ifdef __cplusplus
+#define VIRT_C_LINKAGE "C"
+#define VIRT_API_BEGIN extern "C" {
+#define VIRT_API_END }
+#else
+#define VIRT_C_LINKAGE
+#define VIRT_API_BEGIN
+#define VIRT_API_END
+#endif
+#endif
+
+#include "Dk/Dksystem.h"
+
 #include "plugin/exe_export.h"
 
 #ifdef LONGJMP_DEBUG
@@ -37,8 +51,6 @@
 #define longjmp(buf,val) ldbg_longjmp((buf),(val))
 #endif
 
-#include "Dk/Dksystem.h"
-
 /* These are about to disappear when merge is complete */
 #define PMN_THREADS	/* Activate new threading model */
 #define PMN_LOG		/* Activate new logging */
@@ -69,6 +81,10 @@
 # define LOW_ORDER_FIRST
 #endif
 
+# define _IEEE_FLOATS
+
+VIRT_API_BEGIN
+
 #include "Dk/Dkparam.h"
 #include "Dk/Dktypes.h"
 #include "Dk/Dktrace.h"
@@ -98,6 +114,8 @@
 
 #include "Dk/Dkstubs.h"
 
+VIRT_API_END
+
 /*
  * Localization macros
  */
@@ -120,9 +138,4 @@
 
 #endif /* ENABLE_NLS */
 
-#if defined (MAC_OS_X_VERSION_10_9) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9)
-#include <xmmintrin.h>
-#define __builtin_ia32_loadups(p) _mm_loadu_ps((p))
-#endif
-
 #endif /* _DK_H */
